cas-server-core-authentication/src/main/java/org/jasig/cas/authentication/DefaultAuthenticationSupervisor.java

```diff
@@ -0,0 +1,110 @@
+package org.jasig.cas.authentication;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+/**
+ * This is {@link DefaultAuthenticationSupervisor}.
+ *
+ * @author Misagh Moayyed
+ * @since 4.2.0
+ */
+@Component("defaultAuthenticationSupervisor")
+public final class DefaultAuthenticationSupervisor implements AuthenticationSupervisor {
```

Comment:
This `supervisor` is really a manager/controller of an `authentication transaction`. The authentication transaction is indeed a collection of on going authentication attempts/steps. And at the end of this `transaction` the final `Authentication/Principal` is computed/produced. The transaction step is a higher level which _does_ delegate the work to the lower layer represented by our beloved `AuthenticationManager` facade API. So, proposal would be to rename the current `AuthenticationSupervisor` to `AuthenticationTransactionManager`